### ISSUE TITLE
Scheduled task doesn't run

### ISSUE THREAD

**Issue by meyer59** [laravel/framework#23911]
- Laravel Version: 5.6.16
- PHP Version: PHP 7.2.1-1
- Database Driver & Version: mariadb  Ver 15.1 Distrib 10.1.26-MariaDB
### Description:
Hi
I have followed the steps on the documentation to run scheduled task, i haveset up the cron entry to `* * * * * php /path-to-your-project/artisan schedule:run >> /home/artisan_cron.log 2>&1`

Then in the Kernel.php file i setup my job task: `$schedule->job(new FichierMouvementJob())->when(true);` but nothing is running.
The output in the ` /home/artisan_cron.log` is always "No scheduled commands are ready to run."

I've made some search in the code and the problem seems to be that the `Schedule` instance in the constructor of `ScheduleRunCommand.php` isn't a singleton. It seems that it is a fresh injected instance, thus losing all the events (task).
Replacing  `$this->schedule->dueEvents($this->laravel)`  by `app(Schedule::class)->dueEvents($this->laravel)`  in ScheduleRunCommand::handle() solve the problem and the task and running correctly.

### Steps To Reproduce:
Set a task in Kernel.php, see if it is runing correctly.

**Comment by wilburpowery** [comment]
Not really. Working on an app with lots of jobs and not seeing any problem.

**Comment by wilburpowery** [comment]
What `QUEUE_DRIVER` are you using in your ENV? 
database?, sync?

**Comment by wilburpowery** [comment]
What `QUEUE_DRIVER` are you using in your ENV?
database?, sync?

**Comment by meyer59** [comment]
`QUEUE_DRIVER=database`

**Comment by wilburpowery** [comment]
Did you run the ‘queue:table’ command to generate the table where all your jobs will be stored?

If you don’t have the table, then it makes sense that no jobs are never found.

**Comment by meyer59** [comment]
I have a lot of jobs running fine already. My queue driver is working fine.

**Comment by wilburpowery** [comment]
Great, please consider closing the issue! 😊

**Comment by mverma16** [comment]
@meyer59 Did you find the root cause of the issue and better solution for it? I am facing the same issue while in my application with Laravel 5.5.40

**Comment by meyer59** [comment]
No really. I upgraded the framework to the next release at that time and worked.

**Comment by lk77** [comment]
Hello,

we were calling Artisan::call() inside boot method of one of our provider and it caused the issue.
When doing that, it instantiate commands sooner than intended, before the execution of the schedule method in the console kernel. And since the ScheduleRunCommand has the Scheduler in it's constructor, and it's not up to date with the one edited by the console kernel, jobs are not running.

**Comment by Bolandish** [comment]
@ik

> Hello,
> 
> we were calling Artisan::call() inside boot method of one of our provider and it caused the issue.
> When doing that, it instantiate commands sooner than intended, before the execution of the schedule method in the console kernel. And since the ScheduleRunCommand has the Scheduler in it's constructor, and it's not up to date with the one edited by the console kernel, jobs are not running.

@lk77 You just made my day 👍
